test/plugins/retry429.js

```diff
@@ -0,0 +1,558 @@
+        auth: { username: ME, password: PASSWORD },
+        method: 'GET'
+      };
+      cloudantClient(req, function(err, resp, data) {
+        assert.equal(err.code, 'ECONNRESET');
+        assert.equal(err.message, 'socket hang up');
+        mocks.done();
+        done();
+      });
+    });
+
+    it('successfully retries request on 429 response and returns 200 response', function(done) {
+      if (process.env.NOCK_OFF) {
+        this.skip();
+      }
+
+      var mocks = nock(SERVER)
+          .get('/foo').times(4)
+          .reply(429, {error: 'too_many_requests', reason: 'Too Many Requests'})
+          .get('/foo')
+          .reply(200, {doc_count: 0});
+
+      var cloudantClient = new Client({
+        https: true,
+        retryAttempts: 5,
+        plugin: 'retry'
+      });
+      var req = {
+        url: SERVER + '/foo',
+        auth: { username: ME, password: PASSWORD },
+        method: 'GET'
+      };
+      cloudantClient(req, function(err, resp, data) {
+        assert.equal(err, null);
+        assert.equal(resp.statusCode, 200);
+        assert.ok(data.indexOf('"doc_count":0') > -1);
+        mocks.done();
+        done();
+      });
+    });
+
+    it('fails to retry request on 429 response and returns error', function(done) {
+      if (process.env.NOCK_OFF) {
+        this.skip();
+      }
+
+      var mocks = nock(SERVER)
+          .get('/foo').times(4)
+          .reply(429, {error: 'too_many_requests', reason: 'Too Many Requests'})
+          .get('/foo')
+          .replyWithError({code: 'ECONNRESET', message: 'socket hang up'});
+
+      var cloudantClient = new Client({
+        https: true,
+        retryAttempts: 5,
+        plugin: 'retry'
+      });
+      var req = {
+        url: SERVER + '/foo',
+        auth: { username: ME, password: PASSWORD },
+        method: 'GET'
+      };
+      cloudantClient(req, function(err, resp, data) {
+        assert.equal(err.code, 'ECONNRESET');
+        assert.equal(err.message, 'socket hang up');
+        mocks.done();
+        done();
+      });
+    });
+
+    it('fails to retry request on 429 response and returns 429 response', function(done) {
+      if (process.env.NOCK_OFF) {
+        this.skip();
+      }
+
+      var mocks = nock(SERVER)
+          .get('/foo').times(5)
+          .reply(429, {error: 'too_many_requests', reason: 'Too Many Requests'});
+
+      var cloudantClient = new Client({
+        https: true,
+        retryAttempts: 5,
+        plugin: 'retry'
+      });
+      var req = {
+        url: SERVER + '/foo',
+        auth: { username: ME, password: PASSWORD },
+        method: 'GET'
+      };
+      cloudantClient(req, function(err, resp, data) {
+        assert.equal(err, null);
+        assert.equal(resp.statusCode, 429);
+        assert.ok(data.indexOf('"error":"too_many_requests"') > -1);
+        mocks.done();
+        done();
+      });
+    });
+  });
+
+  describe('with listener only', function() {
```

Comment:
Might be nice to add a test to this lot that pipes the response stream to ensure that the path through the pass through works as expected.